packages/governance/src/typeGuards.js

```diff
@@ -0,0 +1,272 @@
+  },
+  contract: InstanceHandleShape,
+});
+export const ParamChangesQuestionSpecShape = harden({
+  method: 'unranked',
+  issue: ParamChangesIssueShape,
+  positions: ParamChangesPositionsShape,
+  electionType: 'param_change',
+  maxChoices: 1,
+  closingRule: ClosingRuleShape,
+  quorumRule: 'majority',
+  tieOutcome: NoParamChangesPositionShape,
+});
+
+export const ParamChangesQuestionDetailsShape = harden({
+  method: 'unranked',
+  issue: ParamChangesIssueShape,
+  positions: ParamChangesPositionsShape,
+  electionType: 'param_change',
+  maxChoices: 1,
+  closingRule: ClosingRuleShape,
+  quorumRule: 'majority',
+  tieOutcome: NoParamChangesPositionShape,
+  questionHandle: makeHandleShape('Question'),
+  counterInstance: InstanceHandleShape,
+});
+
+const ApiInvocationSpecShape = harden({
+  apiMethodName: M.string(),
+  methodArgs: M.arrayOf(M.any()),
+});
+export const YesApiInvocationPositionShape = ApiInvocationSpecShape;
+export const NoApiInvocationPositionShape = harden({
+  dontInvoke: M.string(),
+});
+export const ApiInvocationPositionsShape = harden([
+  YesApiInvocationPositionShape,
+  NoApiInvocationPositionShape,
+]);
+export const ApiInvocationQuestionSpecShape = harden({
+  method: 'unranked',
+  issue: ApiInvocationSpecShape,
+  positions: ApiInvocationPositionsShape,
+  electionType: 'api_invocation',
+  maxChoices: 1,
+  closingRule: ClosingRuleShape,
+  quorumRule: QuorumRuleShape,
+  tieOutcome: NoApiInvocationPositionShape,
+});
+export const ApiInvocationQuestionDetailsShape = harden({
+  method: 'unranked',
+  issue: ApiInvocationSpecShape,
+  positions: ApiInvocationPositionsShape,
+  electionType: 'api_invocation',
+  maxChoices: 1,
+  closingRule: ClosingRuleShape,
+  quorumRule: QuorumRuleShape,
+  tieOutcome: NoApiInvocationPositionShape,
+  questionHandle: makeHandleShape('Question'),
+  counterInstance: InstanceHandleShape,
+});
+
+const SimpleSpecShape = harden({
+  text: M.string(),
+});
+export const YesSimplePositionShape = harden({ text: M.string() });
+export const NoSimplePositionShape = harden({ text: M.string() });
+export const SimplePositionsShape = harden([
+  YesSimplePositionShape,
+  NoSimplePositionShape,
+]);
+export const SimpleIssueShape = SimpleSpecShape;
+export const SimpleQuestionSpecShape = harden({
+  method: ChoiceMethodShape,
+  issue: SimpleIssueShape,
+  positions: SimplePositionsShape,
+  electionType: M.or('election', 'survey'),
+  maxChoices: M.gte(1),
+  closingRule: ClosingRuleShape,
+  quorumRule: QuorumRuleShape,
+  tieOutcome: NoSimplePositionShape,
+});
+export const SimpleQuestionDetailsShape = harden({
+  method: ChoiceMethodShape,
+  issue: SimpleIssueShape,
+  positions: SimplePositionsShape,
+  electionType: M.or('election', 'survey'),
+  maxChoices: M.gte(1),
+  closingRule: ClosingRuleShape,
+  quorumRule: QuorumRuleShape,
+  tieOutcome: NoSimplePositionShape,
+  questionHandle: makeHandleShape('Question'),
+  counterInstance: InstanceHandleShape,
+});
```

Comment:
AFAICT, the parts in common with `SimpleQuestionSpecShape` are identical. If this is an intentional part of the design, then

```suggestion
export const SimpleQuestionDetailsShape = harden({
  ...SimpleQuestionSpecShape,
  questionHandle: makeHandleShape('Question'),
  counterInstance: InstanceHandleShape,
});
```

Wherever interface extension is an intentional design, this is a good way to express it.

Reply:
Thanks. I was looking for that, and didn't find it.